### ISSUE TITLE
from pipecat.processors.frameworks.langchain import LangchainProcessor,Usage of tools in langchain

### ISSUE THREAD

**Issue by geekofycoder** [pipecat-ai/pipecat#298]
I was trying to implement your interruptible langchain example by moulding it to my usecase.As we can call the tools in OpenAI,similarly I wanted to implement the tools with langchain.Is there a way to use the tools of Langchain in Pipecat?

**Comment by TomTom101** [comment]
You sure can, the processor takes any Langchain Runnable. A chain with tools is basically this:
```
from langchain_core.tools import tool
from langchain_openai import ChatOpenAI

@tool
def multiply(first_int: int, second_int: int) -> int:
    """Multiply two integers together."""
    return first_int * second_int
    
llm = ChatOpenAI(model="gpt-4o").bind_tools([multiply])
```
Then use the `llm` in your chain

`chain = prompt | llm`

Does this help?

**Comment by geekofycoder** [comment]
yes it helps a lot and i think multiple agentic tools can be passed into the chain.Thanks for helping!

**Comment by TomTom101** [comment]
Ideally you close the issue if your questions has been answered.

**Comment by geekofycoder** [comment]
openai.BadRequestError: Error code: 400 - {'error': {'message': "An assistant message with 'tool_calls' must be followed by tool messages responding to each 'tool_call_id'. The following tool_call_ids did not have response messages: call_EuvlW1rYCMs6ne4mELZ3EWnB", 'type': 'invalid_request_error', 'param': 'messages.[5].role', 'code': None}}

This is the error that is coming after I bind the basic multiply and addition function to your interruptible langchain example

**Comment by geekofycoder** [comment]
> Ideally you close the issue if your questions has been answered.

The error occured,you can check in the comment thread

**Comment by TomTom101** [comment]
Guess I should have tried it first :)
I'll check and communicate here! Thanks for the heads-up!

**Comment by geekofycoder** [comment]
> Guess I should have tried it first :) I'll check and communicate here! Thanks for the heads-up!

thats cool...guess the langchain and langgraph tool usage examples are also coming.....that will be really helpful

**Comment by sthitap0** [comment]
@TomTom101 Any updates on the tool call?
I am facing the same issue.

**Comment by TomTom101** [comment]
I am still alive but running on little time currently. I hope to have some space next week to dive into this!

**Comment by 0xtotaylor** [comment]
Bump

**Comment by markbackman** [comment]
Thanks for opening this issue! The `LangchainProcessor` is maintained as a lightweight text-in/text-out bridge for simple Langchain runnable integrations. For advanced features like tool/function calls, role handling, and multi-service workflows, we recommend using Pipecat's native LLM services which have full support for these capabilities.

Closing as part of a housekeeping pass — feel free to reopen if you believe there's still a gap.
